org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/inlayhint/LSPLineContentCodeMining.java

```diff
@@ -91,4 +113,81 @@ private static org.eclipse.jface.text.Position toPosition(Position position, IDo
 		return new org.eclipse.jface.text.Position(start, 1);
 	}
 
+	@Override
+	public final Consumer<MouseEvent> getAction() {
+		return me -> {
+			String title= getLabel();
+			if (title != null && !title.isEmpty()) {
+				findLabelPart(me).ifPresent(labelPart -> {
+					final Command command = labelPart.getCommand();
+					if(command != null && command.getCommand() != null && !command.getCommand().isEmpty()) {
+						ExecuteCommandOptions provider = wrapper.getServerCapabilities().getExecuteCommandProvider();
+						if (provider != null && provider.getCommands().contains(command.getCommand())) {
+							LanguageServerDocumentExecutor.forDocument(document).computeAll((w, ls) -> {
+								if (w == this.wrapper) {
+									return ls.getWorkspaceService()
+											.executeCommand(new ExecuteCommandParams(command.getCommand(), command.getArguments()));
+								}
+								return CompletableFuture.completedFuture(null);
+							});
+						} else  {
+							CommandExecutor.executeCommandClientSide(command, document);
+						}
+					}
+				});
+			}
+		};
+	}
+
+	private Optional<InlayHintLabelPart> findLabelPart(MouseEvent me) {
+		if (inlayHint.getLabel().isRight()) {
+			if (inlayHint.getLabel().getRight().size() == 1) {
```

Comment:
Could we extract inlayHint.getLabel().getRight() to a local variable and replace the three accesses in this method?
```suggestion
			if (inlayHint.getLabel().getRight().size() == 1) {
```

Reply:
Sounds good, did this.